src/test/java/teammates/test/driver/HtmlHelper.java

```diff
@@ -287,4 +304,143 @@ private static boolean isVoidElement(String elementName){
+                      // commentid in URLs
+                      .replaceAll("#" + REGEX_COMMENT_ID, "#\\${comment\\.id}")
+                      // commentid as part of div ids
+                      .replaceAll("responseCommentRow-" + REGEX_COMMENT_ID, "responseCommentRow-\\${comment\\.id}")
+                      .replaceAll("commentBar-" + REGEX_COMMENT_ID, "commentBar-\\${comment\\.id}")
+                      .replaceAll("plainCommentText-" + REGEX_COMMENT_ID, "plainCommentText-\\${comment\\.id}")
+                      // today's date
+                      .replace(TimeHelper.formatDate(now), "${today}")
+                      // date/time now e.g [Thu, 07 May 2015, 07:52 PM] or [Thu, 07 May 2015, 07:52 PM UTC]
+                      .replaceAll(dateTimeNow + REGEX_DISPLAY_TIME, "\\${datetime\\.now}")
+                      // jQuery files
+                      .replace("/js/lib/jquery.min.js", "${lib.path}/jquery.min.js")
+                      .replace("https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js", "${lib.path}/jquery.min.js")
+                      // jQuery-ui files
+                      .replace("/js/lib/jquery-ui.min.js", "${lib.path}/jquery-ui.min.js")
+                      .replace("https://ajax.googleapis.com/ajax/libs/jqueryui/1.10.4/jquery-ui.min.js", "${lib.path}/jquery-ui.min.js")
+                      // admin footer, test institute section
+                      .replaceAll("(?s)<div( class=\"col-md-8\"| id=\"adminInstitute\"){2}>"
+                                              + REGEX_ADMIN_INSTITUTE_FOOTER + "</div>",
+                                  "\\${admin\\.institute}")
+                      // top HTML tag with xmlns defined
+                      // TODO check if this is necessary
+                      .replace("<html xmlns=\"http://www.w3.org/1999/xhtml\">", "<html>")
+                      // noscript is to be cleared
+                      // TODO check if wildcarding this is better; better yet, check if not removing at all works
+                      .replaceFirst("(?s)<noscript>.*</noscript>", "");
+    }
+
+    private static String replaceInjectedValuesWithPlaceholders(String content) {
+        return content.replace("window.location.origin + '/" + Config.STUDENT_MOTD_URL + "';",
+                               "window.location.origin + '/${studentmotd.url}';")
+                      .replace("V" + TP.TEAMMATES_VERSION, "V${version}")
+                      .replace(TP.TEST_STUDENT1_ACCOUNT, "${test.student1}")
+                      .replace(TP.TEST_STUDENT2_ACCOUNT, "${test.student2}")
+                      .replace(TP.TEST_INSTRUCTOR_ACCOUNT, "${test.instructor}")
+                      .replace(TP.TEST_ADMIN_ACCOUNT, "${test.admin}");
+    }
+    
+    /**
+     * This method is only used for testing.
+     */
+    public static String injectImportantValuesForTest(String content) {
```

Comment:
`Important` is a bit subjective. Any way to give a better name? e.g. `ContextDependent`